cmd/axelard/cmd/vald/evm/evm.go

```diff
@@ -453,33 +453,74 @@ func parseTransferKeyConfirmationParams(cdc *codec.LegacyAmino, attributes map[s
 		nil
 }
 
-func (mgr Mgr) validate(rpc rpc.Client, txID common.Hash, confHeight uint64, validateTx func(tx *geth.Transaction, txReceipt *geth.Receipt) bool) bool {
+func getLatestFinalizedBlockNumber(rpc rpc.Client, confHeight uint64) (*big.Int, error) {
+	if rpc.IsMoonbeam() {
+		finalizedBlockHash, err := rpc.ChainGetFinalizedHead(context.Background())
+		if err != nil {
+			return nil, err
+		}
+
+		header, err := rpc.ChainGetHeader(context.Background(), finalizedBlockHash)
+		if err != nil {
+			return nil, err
+		}
+
+		return header.Number.ToInt(), nil
+	}
+
 	blockNumber, err := rpc.BlockNumber(context.Background())
 	if err != nil {
-		mgr.logger.Debug(sdkerrors.Wrap(err, "checking block number failed").Error())
-		// TODO: this error is not the caller's fault, so we should implement a retry here instead of voting against
-		return false
+		return nil, err
 	}
 
+	return big.NewInt(int64(blockNumber - confHeight + 1)), nil
```

Comment:
why +1 here?

Reply:
cuz having 1 confirmation means that the tx is included in the latest block.